src/EnergyPlus/DElightManagerF.cc

```diff
@@ -262,460 +241,354 @@ namespace DElightManagerF {
-
-				// Write each opaque bounding Surface to the DElight input file
-				for ( int isurf = iSurfaceFirst; isurf <= iSurfaceLast; ++isurf ) {
-
-					// Only process "opaque bounding" surface types
-					if ( ( Surface( isurf ).Class == SurfaceClass_Wall ) || ( Surface( isurf ).Class == SurfaceClass_Roof ) || ( Surface( isurf ).Class == SurfaceClass_Floor ) ) {
-
-						// Get the Construction index for this Surface
-						iconstruct = Surface( isurf ).Construction;
-
-						// Is this Surface exposed to the exterior?
-						if ( Surface( isurf ).ExtSolar ) {
-							// Get the index for the outside (i.e., 1st) Material Layer for this Construction
-							iMatlLayer = Construct( iconstruct ).LayerPoint( 1 );
-							// Get the outside visible reflectance of this material layer
-							// (since Construct(iconstruct)%ReflectVisDiffFront always appears to == 0.0)
-							rExtVisRefl = 1.0 - Material( iMatlLayer ).AbsorpVisible;
-						} else {
-							rExtVisRefl = 0.0;
-						}
-
-						// Remove any blanks from the Surface Name for ease of input to DElight
-						cNameWOBlanks = ReplaceBlanksWithUnderscores( Surface( isurf ).Name );
-						gio::write( unit, Format_907 ) << cNameWOBlanks << Surface( isurf ).Azimuth << Surface( isurf ).Tilt << Construct( iconstruct ).ReflectVisDiffBack << rExtVisRefl << Surface( isurf ).Sides;
-
-						// Write out the vertex coordinates for each vertex
-						int const iNumVertices = Surface( isurf ).Sides; // Counter for surface vertices
-						for ( int ivert = 1; ivert <= iNumVertices; ++ivert ) {
-							gio::write( unit, Format_908 ) << Surface( isurf ).Vertex( ivert ).x * M2FT << Surface( isurf ).Vertex( ivert ).y * M2FT << Surface( isurf ).Vertex( ivert ).z * M2FT;
-						}
-
-						// Count each Window hosted by the current opaque bounding Surface
-						iNumWindows = 0;
-						for ( int iwndo = iSurfaceFirst; iwndo <= iSurfaceLast; ++iwndo ) {
-							if ( Surface( iwndo ).Class == SurfaceClass_Window ) {
-								if ( Surface( iwndo ).BaseSurfName == Surface( isurf ).Name ) {
-
-									// Error if window has multiplier > 1 since this causes incorrect illuminance calc
-									if ( Surface( iwndo ).Multiplier > 1.0 ) {
-										ShowSevereError( "Multiplier > 1.0 for window " + Surface( iwndo ).Name + " not allowed since it is in a zone with DElight daylighting." );
-										ErrorsFound = true;
-									}
 
-									// Error if window has a shading device (blind/shade/screen) since
-									// DElight cannot perform dynamic shading device deployment
-									if ( Surface( iwndo ).WindowShadingControlPtr > 0 ) {
-										ShowSevereError( "Shading Device on window " + Surface( iwndo ).Name + " dynamic control is not supported in a zone with DElight daylighting." );
-										ErrorsFound = true;
-									}
+		for ( auto & znDayl : ZoneDaylight ) {
+			if ( znDayl.DaylightMethod == DElightDaylighting ){
+				int const izone = FindItemInList( znDayl.ZoneName, Zone );
```

Comment:
It should only be getting called once since DElightInputGenerator() is being called from GetDaylightingParametersInput(). Just to check I ran a test file and it only hit that line once.